### ISSUE TITLE
[Question] 'sudo: /usr/bin/sudo must be owned by uid 0 and have the setuid bit set'

### ISSUE THREAD

**Issue by junjiequ** [ostreedev/ostree#2142]
I am trying to deploy an customized OSTree commit to my running Fedora CoreOS.

To illustrate what I am doing:

```bash
# On my Fedore CoreOs machine:
$ ls /usr/bin/sudo -lZ
---s--x--x. 2 root root system_u:object_r:sudo_exec_t:s0 182456 Jan  1  1970 /usr/bin/sudo

# On my build machine:
## OSTree repo :fedora-coreos-32.20200618.testing-devel.0-ostree.x86_64.tar 
## is built using cosa assembler against the testing-devel branch of fedora-coreos-config
## The image built from the same OSTree commit could be installed.
tar -C ~/temp-repo/a -xf fedora-coreos-32.20200618.testing-devel.0-ostree.x86_64.tar

mkdir -p ~/temp-repo/b/ ~/temp-repo/c ~/temp-fs/dummy_tree/usr/etc

ostree init --repo=~/temp-repo/b --mode=bare-user

ostree pull-local --repo=~/temp-repo/b ~/temp-repo/b fedora/x86_64/coreos/testing-devel

## I use this as the base os tree
ostree checkout --repo=~/temp-repo/b --require-hardlinks --user-mode 
\--union fedora/x86_64/coreos/testing-devel ~/temp-fs/base_tree/

## then add some customization on top
echo "test" > ~/temp-fs/dymmy_tree/usr/etc/dummy

## make a new commit
ostree commit  --repo=~/temp-repo/b -b a/new/branch --link-checkout-speedup 
\--selinux-policy=~/temp-fs/base_tree  --tree=dir=~/temp-fs/base_tree --tree=dir=~/temp-fs/dummy_tree

## make an archive repo
ostree init --repo=~/temp-repo/c --mode=archive 

## pull the commit into it
ostree pull-local --repo=~/temp-repo/c ~/temp-repo/b a/new/branch

## I didn't expose this repo via http server - by instead, I made a tar file out of it
tar -C ~/temp-repo/c -cf ~/new_commit.tar

## I then deployed new_commit.tar onto the client Fecore CoreOs machine. 

# On my Fedore CoreOs machine - after booted into the new deployment:
## I got this error:
sudo ls
sudo: /usr/bin/sudo must be owned by uid 0 and have the setuid bit set

## looking into the new deploy: 
ls /ostree/deploy/fedora-coreos/deploy/[new deployed commit number]/usr/bin/sudo -lZ
---x--x--x.  2 root root system_u:object_r:sudo_exec_t:s0                    182456 Jan  1  1970 sudo
```

It looks like that the permission on /usr/bin/sudo (possibly the other ones) has been changed. 

The good deploy shows:
```bash
ls /ostree/deploy/fedora-coreos/deploy/[original deploy commit]/usr/bin/sudo -lZ
---s--x--x. 2 root root system_u:object_r:sudo_exec_t:s0 182456 Jan  1  1970 /ostree/deploy/fedora-coreos/deploy/51293dd1ad7571ac5e1a9804c142345afcd6e9bf0912df11752c5cd4f34f5eb9.0/usr/bin/sudo

```

I wonder what I did wrong in these steps causing the setuid bit missing in the new commit?

Could anyone advice please? 

Thanks in advance!

**Comment by jlebon** [comment]
I thought it was the user-mode checkout, but no it seems to work locally. Hmm offhand, it seems fine. You can try adding `ostree ls ${commit} /usr/bin/sudo` at the steps where you create/pull new commits throughout to see at what step it's losing the setuid bit.

**Comment by junjiequ** [comment]
Hey @jlebon, thanks for the reply.

I play with the script again and succeeded with no problem reported initially.
All `ostree ls ${commit} /usr/bin/sudo` commands printed with the correct permissions:
`-04111 0 0 182456 /usr/bin/sudo`

I think I might have some drifts in my dev environment from the code above...  I did play around with it a bit by invoking some commands with `sudo`.... Anyway I don't have any issue now, thanks for your help!

Closing it now.
